cl/corpus_importer/management/commands/update_casenames_wl_dataset.py

```diff
@@ -0,0 +1,360 @@
+                west_case_name,
+                court,
+                date_str,
+                cite1,
+                cite2,
+                docket,
+                volume,
+            ) = row
+
+            clean_docket_num = docket.strip('="').strip('"')
+            if not clean_docket_num:
+                logger.info(f"Row index: {index} - No docket number found.")
+                continue
+
+            date_filed = parse_date(date_str)
+            if not date_filed:
+                logger.info(f"Row index: {index} - No valid date found.")
+                continue
+
+            valid_citations = parse_citations([cite1, cite2])
+
+            if not valid_citations:
+                logger.info(f"Row index: {index} - Missing two citations.")
+                continue
+
+            # Query for possible matches using data from row
+            possible_matches = query_possible_matches(
+                valid_citations=valid_citations,
+                docket_number=clean_docket_num,
+                date_filed=date_filed,
+            )
+
+            if not possible_matches:
+                logger.info(f"Row index: {index} - No matches found.")
+                continue
+
+            matches = []
+            for match in possible_matches:
+                cl_case_name = (
+                    match.cluster.case_name_full
+                    if match.cluster.case_name_full
+                    else match.cluster.case_name
+                )
+                case_name_match = check_case_names_match(
+                    west_case_name, cl_case_name
+                )
+                if case_name_match:
+                    matches.append(match)
+
+            if not possible_matches:
+                logger.info(f"Row index: {index} - No matches found.")
+                continue
+
+            if len(matches) != 1:
+                logger.warning(
+                    f"Row index: {index} - Failed, Matches found: {len(matches)} - Matches: {[cluster.id for cluster in matches]}"
+                )
+                continue
+
+            logger.info(
+                f"Row index: {index} - Match found: {matches[0].cluster_id} - Csv case name: {west_case_name}"
+            )
+
+            if dry_run:
+                # Dry run, don't save anything
+                continue
+
+            # Update case names
+            cluster_updated, docket_updated = update_matched_case_name(
+                matches[0].cluster, west_case_name
+            )
+
+            if cluster_updated:
+                total_clusters_updated = +1
+
+            if docket_updated:
+                total_dockets_updated = +1
+
+            # Add any of the citations if possible
+            add_citations_to_cluster(
+                [
+                    f"{cite.get('volume')} {cite.get('reporter')} {cite.get('page')}"
+                    for cite in valid_citations
+                ],
+                matches[0].cluster_id,
+            )
```

Comment:
reporter or corrected_reporter ?

Reply:
reporter, because we generate a list of dicts with volume, reporter(we get it with corrected_reporter()) and page in parse_citations()